pkgs/development/libraries/science/math/or-tools/default.nix

```diff
@@ -43,14 +84,27 @@ stdenv.mkDerivation rec {
     (cd temp_python/ortools; PYTHONPATH="$python/${python.sitePackages}:$PYTHONPATH" python setup.py install '--prefix=$python')
   '';
 
+  enableParallelBuilding = true;
+
   nativeBuildInputs = [
-    cmake lsb-release swig which zlib python
+    cmake
+    lsb-release
+    swig4
+    which
     ensureNewerSourcesForZipFilesHook
-    python.pkgs.setuptools python.pkgs.wheel
+    python.pkgs.setuptools
+    python.pkgs.wheel
+  ];
+  buildInputs = [
+    zlib
+    bzip2
+    python
   ];
   propagatedBuildInputs = [
-    abseil-cpp gflags glog protobuf cbc
```

Comment:
`abseil-cpp` and `protobuf` are part of the public interface, so should be propagated.

Reply:
I was under the impression that they'd show up in the build closure because the full path would be included in the or-tools derivation even without having them in propagatedBuildInputs, but maybe that's incorrect because of static linking. Thanks for flagging.

Reply:
They do show up in the build closure of or-tools, but they need to be propagated into the build closure of other derivations that use or-tools, in order for these `#include`s to resolve when compiling against the C library.

```console
$ grep '#include' /nix/store/ds0bg62b2a7l3adhy4kgrxn0r356ghch-or-tools-8.1/include/ortools/base/file.h
#include <cstdio>
#include <cstdlib>
#include <string>
#include "absl/status/status.h"
#include "absl/strings/string_view.h"
#include "google/protobuf/descriptor.h"
#include "google/protobuf/io/tokenizer.h"
#include "google/protobuf/message.h"
#include "google/protobuf/text_format.h"
#include "ortools/base/integral_types.h"
#include "ortools/base/logging.h"
```

Reply:
Got it. Done.